pallets/amm/src/lib.rs

```diff
@@ -741,27 +758,30 @@ impl<T: Config> AMM<T::AccountId, AssetId, Balance> for Module<T> {
 	fn execute_buy(transfer: &AMMTransfer<T::AccountId, AssetId, Balance>) -> DispatchResult {
 		let pair_account = Self::get_pair_id(&transfer.asset_sell, &transfer.asset_buy);
 
-		if transfer.discount && transfer.discount_amount > 0 {
-			let hdx_asset = T::HDXAssetId::get();
-			T::Currency::withdraw(hdx_asset, &transfer.origin, transfer.discount_amount)?;
-		}
+		// REVIEW: same note as above re atomicity
+		with_transaction_result(|| {
```

Comment:
Just curious . Does using with_transaction_result have any performance impact vs previous iplmetation without it?

Reply:
It does come with some performance impact, yes, though it should not be very significant if used judiciously.